metaflow/datatools/s3.py

```diff
@@ -1348,32 +1368,245 @@ def _s3op_with_retries(self, mode, **options):
         if self._s3_client_params is not None:
             cmdline.extend(("--s3clientparams", json.dumps(self._s3_client_params)))
 
-        for i in range(S3_RETRY_COUNT + 1):
+        def _get_inject_failures():
+            # list mode does not do retries on transient failures (there is no
+            # SlowDown handling) so we never inject a failure rate
+            if mode == "list":
+                return 0
+            # Otherwise, we cap the failure rate at 90%
+            return min(90, self._s3_inject_failures)
+
+        retry_count = 0  # Number of retries (excluding transient failures)
+        transient_retry_count = 0  # Number of transient retries (per TL retry)
```

Comment:
Maybe I am missing something, but why do we have a distinction between a normal retry and a transient retry?

Reply:
added a very detailed comment.